File: OOP Grouped Quiz 1/Inheritance(Java)/NormalCar.java
class NormalCar {
    String car, color;
    int wheels, weight, engine, price;
    //Null constructor
    public NormalCar(){
        car = "";
        color = "";
        wheels = 0;
        weight = 0;
        engine = 0;
        price = 0;
    }
    //Parameterized constructor
    public NormalCar(String car, String color, int wheels, int weight, int engine, int price){
        this.car = car;
        this.color = color;
        this.wheels = wheels;
        this.weight = weight;
        this.engine = engine;
        this.price = price;
    }
    //Copy constructor
    public NormalCar(NormalCar copy){
        this.car = copy.car;
        this.color = copy.color;
        this.wheels = copy.wheels;
        this.weight = copy.weight;
        this.engine = copy.engine;
        this.price = copy.price;
    }
    //Getter
    void getCar(){
        System.out.println("Car: " + car);
        System.out.println("Color: " + color);
        System.out.println("Wheels: " + wheels);
        System.out.println("Weight: " + weight);
        System.out.println("Engine: " + engine);
        System.out.println("Price: " + price);
    }
    //Setter
    void setCar(String car, String color, int wheels, int weight, int engine, int price){
        this.car = car;
        this.color = color;
        this.wheels = wheels;
        this.weight = weight;
        this.engine = engine;
        this.price = price;
    }
    //Display
    void display(){
        System.out.println("Car: " + car);
        System.out.println("Color: " + color);
        System.out.println("Wheels: " + wheels);
        System.out.println("Weight: " + weight);
        System.out.println("Engine: " + engine);
        System.out.println("Price: " + price);
    } 
}

File: OOP Grouped Quiz 1/Inheritance (CPP)/normalCar.cpp
#include <iostream>
#include <string>
#include "normalCar.h"
using namespace std;

    //null constructor
    normalCar::normalCar()
    {

    }

     //parametrize constructor
    normalCar :: normalCar(string c ,string col ,int w, int wei , int e,int p) 
    {
        car = c;
        colour = col;
        wheels = w;
        weight = wei;
        engine = e;
        price = p;

    }

    //copy constructor
    normalCar::normalCar(const normalCar& userObj)
    {
        car = userObj.car;
        colour = userObj.colour;
        wheels = userObj.wheels;
        weight = userObj.weight;
        engine = userObj.engine;
        price = userObj.price;
    }

    //operator overloading
    ostream& operator<<(ostream& os, normalCar& obj)
    {

    // write obj to stream
    os<<obj.weight<<obj.engine<<obj.price<<obj.wheels<<obj.car <<obj.colour;
    return os;
    }

    istream& operator>>(istream& is, normalCar& obj)
    {
        
    // read obj from stream
    is>>obj.weight>>obj.engine>>obj.price>>obj.wheels>>obj.car >>obj.colour;
    return is;
    }

    // get specifications and show
    void normalCar::getAndshow(normalCar car  ,normalCar colour,  normalCar wheels,normalCar weight,normalCar engine,normalCar price)
      { 
            cout<<"car"<<endl;
            cin>>car.car;
            cout<<"colour"<<endl;
            cin>>colour.colour;
            cout<<"wheels"<<endl;
            cin>>wheels.wheels;
            cout<<"weight"<<endl;
            cin>>weight.weight;
            cout<<"engine?"<<endl;
            cin>>engine.engine;
            cout<<"price?"<<endl;
            cin>>price.price;
            cout<<"car = "<<car.car<<endl<<"colour = "<<colour.colour<<endl<<"weight(lb) = "<<weight.weight<<endl<<"engine = "<<endl<< engine.engine<<endl<<"price = "<<price.price<<endl<<" wheels ="<<wheels.wheels<<endl;
      }

    
    //destructor
    normalCar::~normalCar()
    {
    
    }
